File: src/main/java/cc/hyperium/launch/patching/conflicts/WorldClientTransformer.java
package cc.hyperium.launch.patching.conflicts;

import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.AbstractInsnNode;
import org.objectweb.asm.tree.ClassNode;
import org.objectweb.asm.tree.IntInsnNode;
import org.objectweb.asm.tree.MethodNode;

import java.util.Iterator;

public class WorldClientTransformer implements ConflictTransformer {
    @Override
    public String getClassName() {
        return "net.minecraft.client.multiplayer.WorldClient";
    }

    @Override
    public ClassNode transform(ClassNode original) {
        for (MethodNode node : original.methods) {
            if (node.name.equals("doVoidFogParticles")) {
                Iterator<AbstractInsnNode> iterator = node.instructions.iterator();
                while (iterator.hasNext()) {
                    AbstractInsnNode insn = iterator.next();
                    if (insn instanceof IntInsnNode
                            && insn.getOpcode() == Opcodes.SIPUSH
                            && ((IntInsnNode) insn).operand == 1000) {
                        ((IntInsnNode) insn).operand = 500;
                    }
                }
            }
        }
        return original;
    }
}

File: src/main/kotlin/cc/hyperium/launch/patching/conflicts/WorldClientTransformer.kt
package cc.hyperium.launch.patching.conflicts

import org.objectweb.asm.tree.ClassNode
import org.objectweb.asm.tree.IntInsnNode

class WorldClientTransformer : ConflictTransformer {
    override fun getClassName() = "bdb"

    override fun transform(original: ClassNode): ClassNode {
        original.methods.find {
            it.name == "doVoidFogParticles"
        }?.apply {
            for (insn in instructions.iterator()) {
                if (insn is IntInsnNode && insn.operand == 1000) {
                    insn.operand = 500
                    break
                }
            }
        }

        return original
    }
}
